lib/daru/vector.rb

```diff
@@ -220,6 +220,10 @@ def [](*indexes)
 
         return result
       else
+      raise TypeError, "Invalid index type #{location.inspect}.\
+        \nUsage: v[:a, :b] gives elements with keys :a and :b for vector v." \
+        if location.is_a? Array
+
```

Comment:
Works. Can you rebase into a single commit?

Also I'm a bit tied down currently with the SciRuby GSOC application so please pardon me for late replies. I'll back to you whenever possible.